lib/saluki-components/src/destinations/dsd_stats/mod.rs

```diff
@@ -0,0 +1,220 @@
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl DogStatsDAPIHandler {
+    async fn stats_handler(State(state): State<DogStatsDAPIHandlerState>) -> (StatusCode, String) {
+        let (oneshot_tx, oneshot_rx) = tokio::sync::oneshot::channel();
+
+        if let Err(e) = state.tx.try_send(oneshot_tx) {
+            return (
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Failed to send stats: {}", e),
+            );
+        }
+
+        match oneshot_rx.await {
+            Ok(stats) => {
+                info!("Stats on received events: {:?}", stats);
+                (StatusCode::OK, serde_json::to_string(&stats).unwrap())
+            }
+            Err(e) => (
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Failed to receive stats: {}", e),
+            ),
+        }
+    }
+}
+
+impl APIHandler for DogStatsDAPIHandler {
+    type State = DogStatsDAPIHandlerState;
+
+    fn generate_initial_state(&self) -> Self::State {
+        self.state.clone()
+    }
+
+    fn generate_routes(&self) -> Router<Self::State> {
+        Router::new().route("/dogstatsd/stats", get(Self::stats_handler))
+    }
+}
+
+impl DogStatsDStatisticsConfiguration {
+    /// Creates a new 'DogStatsDStatisticsConfiguration' from the given configuration.
+    pub fn from_configuration(_: &GenericConfiguration) -> Result<Self, GenericError> {
+        let (tx, rx) = mpsc::channel(100);
+        let state = DogStatsDAPIHandlerState { tx: Arc::new(tx) };
+        let handler = DogStatsDAPIHandler { state };
+
+        Ok(Self {
+            api_handler: handler,
+            rx: Arc::new(Mutex::new(rx)),
+            max_metrics_per_payload: default_max_metrics_per_payload(),
+        })
+    }
+
+    /// Returns an API handler for DogStatsD API.
+    pub fn api_handler(&self) -> DogStatsDAPIHandler {
+        self.api_handler.clone()
+    }
+}
+
+#[async_trait]
+impl DestinationBuilder for DogStatsDStatisticsConfiguration {
+    fn input_event_type(&self) -> EventType {
+        EventType::Metric
+    }
+
+    async fn build(&self, _context: ComponentContext) -> Result<Box<dyn Destination + Send>, GenericError> {
+        let rx = Arc::clone(&self.rx);
+        Ok(Box::new(DogStatsDStats {
+            rx,
+            stats: Stats {
+                metrics_received: HashMap::new(),
+            },
+        }))
+    }
+}
+
+impl MemoryBounds for DogStatsDStatisticsConfiguration {
+    fn specify_bounds(&self, builder: &mut MemoryBoundsBuilder) {
+        builder
+            .minimum()
+            .with_single_value::<DogStatsDStatisticsConfiguration>("configuration struct");
+
+        builder.firm().with_expr(UsageExpr::constant("api handler state", 48));
+        builder.firm().with_expr(UsageExpr::constant("destination state", 24));
+        builder
+            .firm()
+            // Capture the size of the "split re-encode" buffers in the request builders, which is where we keep owned
```

Comment:
I think we can remove this (split re-encode buffer) since it doesn't apply here I believe.